### PR TITLE
Add line numbers to the diff view

### PR DESCRIPTION
**Pull request by txsmith**
screenshot from 2016-05-20 16 21 13

### CODE REVIEW

File: src/pull-request/pull-request-diff-highlight.html [hunk 1]
```diff
@@ -70,6 +97,50 @@
...
         },
         file: String,
+        linesChanged: Object
+      },
+      observers: [
+        '_observeLinesChanged(linesChanged, code)'
+      ],
+      _observeLinesChanged: function(linesChanged, code) {
+        var lines = code.split('\n')
+        // The first and list line are always empty and should be ignored.
```
Comment (eanker) [inline src/pull-request/pull-request-diff-highlight.html]: list --> last, just a textual mistake

**Comment by eanker** [comment]
The code and result seem nice.

**Comment by TimvdLippe** [comment]
I wonder if we can make the amount of DOM mutations lower. This current solution has a lot of single DOM operations, while we could also flush it once as a whole. So maybe create a String of `<span>` and then append with `innerHTML`. This then only happens once, instead of for each time remove and add `<span>` separately.

**Comment by txsmith** [comment]
Is there any way we can measure these things in a reliable way?

**Comment by TimvdLippe** [comment]
You could artificially increase the number of times you perform these operations by putting it in a for-loop. For large PR's (e.g. several K line number changes), 1000 times should be sufficient to test.

**Comment by GijsWeterings** [comment]
```
console.time("testSomeAction");
    /* the action here */
console.timeEnd("testSomeAction");
```

^ with many repetitions (for loop). Timings are also visible in the timeline view in devtools

**Comment by txsmith** [comment]
Perhaps I was not very clear in my original question. I'm curios whether there is a good benchmarking library for things like this.

**Comment by GijsWeterings** [comment]
There is [benchmark.js](https://www.npmjs.com/package/benchmark), but the console.time feature is easier if you just want to run a few experiments

**Comment by TimvdLippe** [comment]
Performant, working, yay! LGTM

**Pull request merged**